workflow-templates/executor-ci.yml

```diff
@@ -1,17 +1,193 @@
+        uses: jitterbit/get-changed-files@v1
+        continue-on-error: true
+      - id: set-matrix
+        run: bash scripts/get-all-test-paths.sh
+        env:
+          CHANGED_FILES: ${{ steps.changed-files.outputs.all }}
+      - id: debug
+        run: echo ${{ steps.set-matrix.outputs.matrix }}
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+
+  linting:
+    needs: [prep-testbed]
+    runs-on: ubuntu-latest
+    if: ${{ needs.prep-testbed.outputs.matrix != '[]' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        executor-path: ${{fromJson(needs.prep-testbed.outputs.matrix)}}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: "3.7"
+      - name: Install requirements
+        run: pip install black flake8 isort
+      - name: Flake8 linting
+        run: flake8 ${{ matrix.executor-path }}
+      - name: Check imports using isort
+        run: isort --check-only ${{ matrix.executor-path }}
+      - name: Black check formatting
+        run: black --check ${{ matrix.executor-path }}
+
+  unit-tests:
+    needs: [prep-testbed]
+    runs-on: ubuntu-latest
+    if: ${{ needs.prep-testbed.outputs.matrix != '[]' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        executor-path: ${{fromJson(needs.prep-testbed.outputs.matrix)}}
+    defaults:
+      run:
+        working-directory: ${{ matrix.executor-path }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: "3.7"
+      - name: Pre-test script
+        run: |
+          if [[ -f "tests/pre_test.sh" ]]; then
+            bash tests/pre_test.sh; fi
+      - name: Install latest stable 2.x jina version
+        run: |
+          JINA_VERSION=$(curl -L -s "https://pypi.org/pypi/jina/json" \
+            |  jq  -r '.releases | keys | .[]
+              | select(contains("dev") | not)
+              | select(startswith("2."))' \
+            | sort -V | tail -1)
+          pip install git+https://github.com/jina-ai/jina.git@v${JINA_VERSION}
+      - name: Python requirements
+        run: pip install -r requirements.txt -r tests/requirements.txt
+      - name: Create Dockerfile, if needed
+        run: |
+          if [[ ! -f "Dockerfile" ]]; then
+            docker run --entrypoint="normalizer" -v "$(pwd):/workspace" \
+            jinahub/hubble-normalizer:v0.1.0 . --jina-version=2
+            sudo chown -R $(id -u ${USER}):$(id -g ${USER}) . ; fi
+      - name: Run unit tests
+        run: pytest -s -v -m "not gpu"
+      # - name: Print image size
+      #   run: |
+      #     IMG_NAME=$(echo "${PWD##*/}" | tr '[:upper:]' '[:lower:]')
+      #     docker inspect $IMG_NAME | jq '.[0].Size' | numfmt --to=iec --suffix=B
+
+  # @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+  # @@@@@@@ ⚠️ DEPRECATED ⚠️ @@@@@@@@@
+  # @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
```

Comment:
I think when we add this CI/CD for single executor repos, we should remove this and make sure any newly created repo does not depend on it.

Reply:
Will adapt after trying it out on one executor repo jina-ai/executor-simpleindexer#3